### ISSUE TITLE
Ommission of semicolon prior to jsdoc causes Missing JSDoc comment error

### ISSUE THREAD

**Issue by mjniday** [eslint/eslint#9298]
**Tell us about your environment**

* **ESLint Version:**
3.17.1
* **Node Version:**
8.4.0
* **npm Version:**
5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "parser"  : "babel-eslint",
  "extends" : [
    "standard",
    "standard-react"
  ],
  "env"     : {
    "browser" : true
  },
  "globals" : {
    "Action"       : false,
    "__DEV__"      : false,
    "__PROD__"     : false,
    "__DEBUG__"    : false,
    "__COVERAGE__" : false,
    "__BASENAME__" : false,
    "React"        : true,
    "ReactDOM"     : true,
    "$"            : true,
    "it"           : true,
    "expect"       : true,
    "should"       : true,
    "describe"     : true,
    "beforeEach"   : true,
    "afterEach"    : true,
    "before"       : true,
    "after"        : true,
    "sinon"        : true,
    "context"      : true
  },
  "plugins": [
    "mocha"
  ],
  "rules": {
    "mocha/no-exclusive-tests": "error",
    "react/no-unused-prop-types": 0,
    "semi" : [2, "never"],
    "max-len": [2, 120, 2],
    "no-duplicate-imports": 0,
    "generator-star-spacing": 0,
    "react/jsx-boolean-value": 0,
    "require-jsdoc": 2,
    "valid-jsdoc": [2, {
      "requireParamDescription": false,
      "requireReturnDescription": false
    }]
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue.**
It seems that any jsdoc preceded by an object without a semicolon will cause eslint to think the jsdoc is missing. 

```js

const itemPropTypes = {
  
}

/**
 * Decorator that provides access to an API-backed item
 *
 * @param   {string} itemType    Type of resource (e.g. 'contact')
 * @param   {Object} itemOptions
 * @param   {string} itemOptions.getId - Function that takes props and returns the 'itemId'
 * @param   {string} itemOptions.propsWrapper - Key in which to wrap itemProps.
 * @param   {bool} itemOptions.preventFetch - Do not fetch on mount.
 *
 * @returns {function}
 */
function item (itemType, itemOptions = {}) { ... }



```

**What did you expect to happen?**
Expected the jsdoc to be valid and not have any linter errors. There are a number of these in the project, and it seems all of them are fixed by adding a preceding semicolon to the file.

**What actually happened? Please include the actual, raw output from ESLint.**
`error  Missing JSDoc comment  require-jsdoc`

**Comment by platinumazure** [comment]
We've made some changes to comment attachment with ESLint 4.x. I [can't reproduce this](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IHJlcXVpcmUtanNkb2M6IDIgKi9cclxuXHJcbmNvbnN0IGl0ZW1Qcm9wVHlwZXMgPSB7XHJcbiAgXHJcbn1cclxuXHJcbi8qKlxyXG4gKiBEZWNvcmF0b3IgdGhhdCBwcm92aWRlcyBhY2Nlc3MgdG8gYW4gQVBJLWJhY2tlZCBpdGVtXHJcbiAqXHJcbiAqIEBwYXJhbSAgIHtzdHJpbmd9IGl0ZW1UeXBlICAgIFR5cGUgb2YgcmVzb3VyY2UgKGUuZy4gJ2NvbnRhY3QnKVxyXG4gKiBAcGFyYW0gICB7T2JqZWN0fSBpdGVtT3B0aW9uc1xyXG4gKiBAcGFyYW0gICB7c3RyaW5nfSBpdGVtT3B0aW9ucy5nZXRJZCAtIEZ1bmN0aW9uIHRoYXQgdGFrZXMgcHJvcHMgYW5kIHJldHVybnMgdGhlICdpdGVtSWQnXHJcbiAqIEBwYXJhbSAgIHtzdHJpbmd9IGl0ZW1PcHRpb25zLnByb3BzV3JhcHBlciAtIEtleSBpbiB3aGljaCB0byB3cmFwIGl0ZW1Qcm9wcy5cclxuICogQHBhcmFtICAge2Jvb2x9IGl0ZW1PcHRpb25zLnByZXZlbnRGZXRjaCAtIERvIG5vdCBmZXRjaCBvbiBtb3VudC5cclxuICpcclxuICogQHJldHVybnMge2Z1bmN0aW9ufVxyXG4gKi9cclxuZnVuY3Rpb24gaXRlbSAoaXRlbVR5cGUsIGl0ZW1PcHRpb25zID0ge30pIHsgfSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==) in our demo.

Can you give this a try with a more recent ESLint version?

**Comment by mjniday** [comment]
@platinumazure Thanks for the quick reply. After trying with ESlint 4.x this seems to have resolved itself.
